components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/IdentityProviderManager.java

```diff
@@ -850,7 +870,7 @@ public void addResidentIdP(IdentityProvider identityProvider, String tenantDomai
         identityProvider.setFederatedAuthenticatorConfigs(IdentityApplicationManagementUtil
                 .concatArrays(identityProvider.getFederatedAuthenticatorConfigs(), federatedAuthenticatorConfigs));
 
-        IdentityProviderProperty[] idpProperties = new IdentityProviderProperty[2];
+        IdentityProviderProperty[] idpProperties = new IdentityProviderProperty[3];
```

Comment:
Can we make this 3 (magic number) a descriptive constant
e.g. SAML_IDP_PROPERTIES_SIZE or something along